### ISSUE TITLE
Service Fabric Setup Configuration

### ISSUE THREAD

**Issue by TrevorVonSeggern** [terraform-providers/terraform-provider-azurerm#1948]
I'm trying to create a working Service Fabric cluster in Azure. The service fabric resource only creates the service fabric management part, ie not the vm scale sets, or networking resources.

How do I create a working SF cluster with all required resources with a Terraform script?

**Comment by TrevorVonSeggern** [comment]
No, I don't think that's the expected behavior. When I first created the resource I waited for like half an hour for the cluster to be ready, of course nothing happened except that it created the one sf resource. I've spent a non-trivial amount of time trying to get the configuration right with all the correct resources using terraform, without anything working. The documentation out there is OK for creating a sf cluster with powershell or arm templates, but so far I haven't found anything on how to create one using terraform or how to set one up using only resources (without looking through arm templates). The least expectation is that there is an example for a correct config, somewhere. Otherwise using this TF resource is next to impossible for developers that don't have an intimate knowledge of the inner workings of azure.

**Comment by steve-hawkins** [comment]
@TrevorVonSeggern forgive me I didn't mean to trivialise the effort involved in creating a Service Fabric cluster, quite the opposite.  I have spent the past few months writing Terraform for varying cluster types and unfortunately with the specification and potential capabilities available it will involve in-depth knowledge of Azure VMSS, extensions, storage, networking, security and potentially more based on your requirements.

This resource has only recently been added to the Terraform provider and is still being developed. 
 I agree an example would be a good addition to the documentation, but maybe in a sample Git repo of its own would be a better place as it will be quite long and involved.

The new [Service Fabric Mesh](https://azure.microsoft.com/en-gb/blog/azure-service-fabric-mesh-is-now-in-public-preview/) is an Azure managed Service Fabric cluster and may be a better option if you do not require customisation to the hosts.

**Comment by tombuildsstuff** [comment]
hi @TrevorVonSeggern 

Thanks for opening this issue - apologies for the delayed response here!

As mentioned above [the `azurerm_service_fabric_cluster` resource](https://www.terraform.io/docs/providers/azurerm/r/service_fabric_cluster.html) only creates the Service Fabric Cluster within Azure (since this is the way the Service Fabric API is designed). Whilst Terraform supports all the individual components required for Service Fabric (e.g. VM Scale Sets/Extensions for building the Compute nodes / Load Balancers etc) - unfortunately we don't have a sample for spinning up all of the dependencies for a Service Fabric Cluster at this time - but as @steve-hawkins has mentioned this should be possible to port from the documentation/an ARM Template.

Again as @steve-hawkins has mentioned, the newer Service Fabric Mesh could be a better fit for your use-case if you're looking for Azure to manage all of these resources for you (rather than managing the Service Fabric nodes yourself) - but we don't currently support this in Terraform. If you're looking for Terraform to support this please [open a Feature Request for this](https://github.com/terraform-providers/terraform-provider-azurerm/issues/new?template=Feature_Request.md) - but since this question appears to be resolved I'm going to close this issue for the moment.

Thanks!
